GitExtUtils/FileLogging/LogFileCleaner.cs

```diff
@@ -0,0 +1,34 @@
+﻿using System;
+using System.IO;
+using System.Linq;
+using System.Threading;
+
+namespace GitExtUtils.FileLogging
+{
+    public class LogFileCleaner
+    {
+        private readonly LogFilenameCreator _logFilenameCreator;
+
+        public LogFileCleaner(LogFilenameCreator logFilenameCreator)
+        {
+            _logFilenameCreator = logFilenameCreator;
+        }
+
+        public void Run()
+        {
+            WaitToPreventAddedStartupTime();
+
+            var deleteThreshold = DateTime.Now - TimeSpan.FromDays(30);
+
+            new DirectoryInfo(_logFilenameCreator.LogPath)
+                .GetFileSystemInfos(_logFilenameCreator.GetCleaningFilenamePattern())
+                .Where(x => x.CreationTime < deleteThreshold)
+                .ForEach(x => x.Delete());
```

Comment:
❗️  This may be randomly failing for a multitude of reasons, so it must be wrapped in `try..catch`

Reply:
Sure, we can also make it a thread and have it called every x hours. My initial judgement was to just do this at startup ;)